cadl-tests/src/main/java/com/cadl/odata/OdataAsyncClient.java

```diff
@@ -0,0 +1,88 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+// Code generated by Microsoft (R) AutoRest Code Generator.
+
+package com.cadl.odata;
+
+import com.azure.core.annotation.Generated;
+import com.azure.core.annotation.ReturnType;
+import com.azure.core.annotation.ServiceClient;
+import com.azure.core.annotation.ServiceMethod;
+import com.azure.core.exception.ClientAuthenticationException;
+import com.azure.core.exception.HttpResponseException;
+import com.azure.core.exception.ResourceModifiedException;
+import com.azure.core.exception.ResourceNotFoundException;
+import com.azure.core.http.rest.PagedFlux;
+import com.azure.core.http.rest.RequestOptions;
+import com.azure.core.util.BinaryData;
+import com.cadl.odata.implementation.OdataOpsImpl;
+
+/** Initializes a new instance of the asynchronous OdataClient type. */
+@ServiceClient(builder = OdataClientBuilder.class, isAsync = true)
+public final class OdataAsyncClient {
+    @Generated private final OdataOpsImpl serviceClient;
+
+    /**
+     * Initializes an instance of OdataAsyncClient class.
+     *
+     * @param serviceClient the service client implementation.
+     */
+    @Generated
+    OdataAsyncClient(OdataOpsImpl serviceClient) {
+        this.serviceClient = serviceClient;
+    }
+
+    /**
+     * The list operation.
+     *
+     * <p><strong>Query Parameters</strong>
+     *
+     * <table border="1">
+     *     <caption>Query Parameters</caption>
+     *     <tr><th>Name</th><th>Type</th><th>Required</th><th>Description</th></tr>
+     *     <tr><td>select</td><td>String</td><td>No</td><td>The select parameter</td></tr>
+     *     <tr><td>expand</td><td>String</td><td>No</td><td>The expand parameter</td></tr>
+     *     <tr><td>filter</td><td>String</td><td>No</td><td>The filter parameter</td></tr>
+     *     <tr><td>orderby</td><td>String</td><td>No</td><td>The orderBy parameter</td></tr>
```

Comment:
Since the convenience method is not supported for pageable yet, the correct casing only reflects in the description "The orderBy parameter".